engine.py

```diff
@@ -97,8 +97,63 @@ def get_knight_moves(self, r, c, moves):
 
 
 	def get_king_moves(self, r, c, moves):
- 		##TODO
- 		pass
+         """
+			Calculates all possible kings moves for a given color (light or dark)
+			and appends them to a list
+
+			input parameter(s):
+			r     --> starting row (int)
+			c     --> starting colum (int)
+			moves --> possible moves container (list)
+
+			return parameter(s):
+			None
+		"""
+         ##TODO
+         if self.light_to_move:
+             if r == 7:#change the iteration of the for loop to make sure moves does'nt exceed board
+                 f = 2   
+             else: f = 3 
+             
+             if c != 0:#do not run if we are in the first column
+                 for i in range(f):#check valid moves one column behind the current column
+                        if self.board[r-1+i][c-1] == "  " or self.board[r-1+i][c-1][1] == "d":
+                            moves.append(Move((r, c), (r-1+i, c-1), self.board))
+                        
+             for i in range(f):#check valid moves on the current column
+                    if self.board[r-1+i][c] == "  " or self.board[r-1+i][c][1] == "d":  
+                        moves.append(Move((r, c), (r-1+i, c), self.board)) 
+                        
+             if c != 7:#do not run if we are in the last column    
+               for i in range(f):#check valid moves one column after the current column
+                    if self.board[r-1+i][c+1] == "  " or self.board[r-1+i][c+1][1] == "d": 
+                        moves.append(Move((r, c), (r-1+i, c+1), self.board))           
+         else:
+             
+             if c == 7:#change the iteration of the for loop to make sure moves does'nt exceed board
+                 f = 2    
+             else: f=3
+             
+             b = c-1#start a column behind   
+             if r != 0:#do not run if we are in the first row
+                for i in range(f):#check for valid moves a row behind the current row
+                    if self.board[r-1][b+i] == "  " or self.board[r-1][b+i][1] == "l": 
+                        if b+i<0:
+                            continue
+                        else:moves.append(Move((r, c), (r-1, b+i), self.board))
+                        
+             for i in range(f):#check for valid moves in the current row
+                    if self.board[r][b+i] == "  " or self.board[r][b+i][1] == "l":
+                        if b+i<0:
+                            continue
+                        else:moves.append(Move((r, c), (r, b+i), self.board)) 
+                        
+             if r != 7:#do not run if we are in the last row
+               for i in range(f):#check for valid moves a row after the current row
+                    if self.board[r+1][b+i] == "  " or self.board[r+1][b+i][1] == "l":
+                        if b+i<0:
+                            continue
+                        else:moves.append(Move((r, c), (r+1, b+i), self.board)) 
```

Comment:
To be consistent with the rest of us, kindly use single tab for all indents and not spaces; It's throwing errors all over the place

Reply:
ok sir, i have corrected it.

Reply:
It's still showing up as spaces not tabs. Please try resolving this with your group leader